core/src/main/java/org/elasticsearch/rest/action/admin/indices/create/RestCreateIndexAction.java

```diff
@@ -52,6 +56,12 @@ public void handleRequest(final RestRequest request, final RestChannel channel,
         createIndexRequest.updateAllTypes(request.paramAsBoolean("update_all_types", false));
         createIndexRequest.timeout(request.paramAsTime("timeout", createIndexRequest.timeout()));
         createIndexRequest.masterNodeTimeout(request.paramAsTime("master_timeout", createIndexRequest.masterNodeTimeout()));
-        client.admin().indices().create(createIndexRequest, new AcknowledgedRestListener<CreateIndexResponse>(channel));
+        createIndexRequest.waitForActiveShards(ActiveShardCount.parseString(request.param("wait_for_active_shards")));
+        client.admin().indices().create(createIndexRequest, new AcknowledgedRestListener<CreateIndexResponse>(channel) {
+            @Override
+            public void addCustomFields(XContentBuilder builder, CreateIndexResponse response) throws IOException {
```

Comment:
This is not as easily achieved because for other cases, adding the custom fields requires information in the request object as well as the context of the rest request, so leaving it as is for now as discussed with @ywelsch